tf-modules/cumulus/ecs_cluster.tf

```diff
@@ -220,45 +220,96 @@ data "aws_efs_mount_target" "ecs_cluster_instance" {
 }
 
 locals {
-  ecs_instance_autoscaling_cf_template_config = {
+  ecs_instance_autoscaling_user_data_config = {
     cluster_name              = aws_ecs_cluster.default.name
     container_stop_timeout    = var.ecs_container_stop_timeout,
     docker_hub_config         = var.ecs_docker_hub_config,
-    docker_volume_size        = var.ecs_cluster_instance_docker_volume_size,
     docker_volume_create_size = var.ecs_cluster_instance_docker_volume_size - 1,
     efs_dns_name              = var.ecs_efs_config == null ? null : data.aws_efs_mount_target.ecs_cluster_instance[0].dns_name,
     efs_mount_point           = var.ecs_efs_config == null ? null : var.ecs_efs_config.mount_point,
-    image_id                  = var.ecs_cluster_instance_image_id,
     include_docker_cleanup_cronjob = var.ecs_include_docker_cleanup_cronjob,
-    instance_profile          = aws_iam_instance_profile.ecs_cluster_instance.arn,
-    instance_type             = var.ecs_cluster_instance_type,
-    key_name                  = var.key_name,
-    min_size                  = var.ecs_cluster_min_size,
-    desired_capacity          = var.ecs_cluster_desired_size,
-    max_size                  = var.ecs_cluster_max_size,
     region                    = data.aws_region.current.name
-    security_group_ids        = compact(concat(
-      [
-        aws_security_group.ecs_cluster_instance.id,
-        var.elasticsearch_security_group_id,
-        var.rds_security_group
-      ],
-      var.ecs_custom_sg_ids
-    ))
-    subnet_ids                = var.ecs_cluster_instance_subnet_ids,
     task_reaper_object        = aws_s3_bucket_object.task_reaper
   }
+
+  security_group_ids = compact(concat(
+    [
```

Comment:
The ecs autoscaling group was done with cloudformation which used launch configuration, and since terraform supports the resources, I migrate to use terraform to created all the resources.